ts/features/bonus/bpd/saga/orchestration/onboarding/enrollToBpd.ts

```diff
@@ -0,0 +1,74 @@
+import { Either, left, right } from "fp-ts/lib/Either";
+import { NavigationActions } from "react-navigation";
+import { SagaIterator } from "redux-saga";
+import { call, Effect, put, race, select, take } from "redux-saga/effects";
+import { ActionType, getType } from "typesafe-actions";
+import { navigationHistoryPop } from "../../../../../../store/actions/navigationHistory";
+import { navigationCurrentRouteSelector } from "../../../../../../store/reducers/navigation";
+import { SagaCallReturnType } from "../../../../../../types/utils";
+import {
+  navigateToBpdOnboardingEnrollPaymentMethod,
+  navigateToBpdOnboardingLoadActivate
+} from "../../../navigation/action";
+import BPD_ROUTES from "../../../navigation/routes";
+import {
+  bpdOnboardingCancel,
+  bpdEnrollUserToProgram
+} from "../../../store/actions/onboarding";
+
+export function* enrollUserToBpd(): Generator<
+  Effect,
+  Either<Error, boolean>,
+  any
+> {
+  yield put(bpdEnrollUserToProgram.request());
+  const enrollUserResults: ActionType<
+    | typeof bpdEnrollUserToProgram.success
+    | typeof bpdEnrollUserToProgram.failure
+  > = yield take([
+    getType(bpdEnrollUserToProgram.success),
+    getType(bpdEnrollUserToProgram.failure)
+  ]);
+  return enrollUserResults.type === getType(bpdEnrollUserToProgram.success)
+    ? right<Error, boolean>(enrollUserResults.payload.enabled)
+    : left<Error, boolean>(enrollUserResults.payload);
```

Comment:
Could this block can be replaced with this shortcut?
`const enrolled = yield call(enrollCitizen);`
note `enrollCitizen` returns false if the citizen is not enrolled and even if the request fails

Reply:
I need to distinguish between the failure (Error) and the success with the result (true,false) because I've two different flow (retry or continue the flow), for this reason i need an`Either<Error,boolean>`

Reply:
@fabriziofff Could these changes make that util more flexible?
fd0a986903b280c0a3807a150c3c0f659fca09d1
Instead of a boolean now it returns `right(value)` on successfully case, `left(Error)` in failure ones

Reply:
I've cherry picked your commit and refactored the functions with the new reusable code.